DEVELOPMENT_GUIDE.md

```diff
@@ -21,93 +17,76 @@ This will start a new Gitpod workspace, and immediately kick off a build of the
-can be found [here](https://black.readthedocs.io/en/stable/editor_integration.html))
- 
-##### (workaround) Integrating Black directly in your favorite IDE
-Since black is installed in virtualenv, when you follow [this instruction](https://black.readthedocs.io/en/stable/editor_integration.html), `which black` might give you this
+**Note**: `pyenv` currently only supports macOS and Linux. If you are a
+Windows users, consider using [pipenv](https://docs.pipenv.org/).
 
-```bash
-(samcli37) $ where black
-/Users/<username>/.pyenv/shims/black
-```
+|    | `venv`   | `pyenv`      |
+| -- | -------- | ------------ |
+| Pick if you want ... | Easy setup | You want to easily develop and test SAM CLI in different Python versions |
+| Setup | None | `curl -L https://github.com/pyenv/pyenv-installer/raw/master/bin/pyenv-installer \| bash` |
+| Install a Python version (3.8.9 as example) | N/A | `pyenv install 3.8.9` |
+| Create a virtual environment | `python3 -m venv .venv` | `pyenv virtualenv 3.8.9 samcli38` (you can rename `samcli38` to something else) |
+| Activate the virtual environment | `source .venv/bin/activate` | `pyenv activate samcli38` |
 
-However, IDEs such PyChaim (using FileWatcher) will have a hard time invoking `/Users/<username>/.pyenv/shims/black` 
-and this will happen:
 
-```
-pyenv: black: command not found
 
-The `black' command exists in these Python versions:
-  3.7.2/envs/samcli37
-  samcli37
-``` 
+**For convenience, below we assume the virtual environment is called `samcli38`.**
 
-A simple workaround is to use `/Users/<username>/.pyenv/versions/samcli37/bin/black` 
-instead of `/Users/<username>/.pyenv/shims/black`.
+### 2. Initialize dependencies and create `samdev` available in `$PATH`
 
-#### Pre-commit
-If you don't wish to manually run black on each pr or install black manually, we have integrated black into git hooks through [pre-commit](https://pre-commit.com/).
-After installing pre-commit, run `pre-commit install` in the root of the project. This will install black for you and run the black formatting on
-commit.
+Clone the AWS SAM CLI repository to your local machine if you haven't done that yet.
 
-### 3. Activate Virtualenv
-
-Virtualenv allows you to install required libraries outside of the
-Python installation. A good practice is to setup a different virtualenv
-for each project. [pyenv](https://github.com/pyenv/pyenv) comes with a
-handy plugin that can create virtualenv.
+```sh
+# Using SSH
+$ git clone git@github.com:aws/aws-sam-cli.git
+# Using HTTPS
+$ git clone https://github.com/aws/aws-sam-cli.git
+```
 
-Depending on the python version, the following commands would change to
-be the appropriate python version.
+(make sure you have virtual environment activated)
 
-1.  `pyenv virtualenv 3.7.2 samcli37`
-2.  `pyenv activate samcli37` for Python3.7
+```sh
+(samcli38) $ cd aws-sam-cli
```

Comment:
The prompt noise makes it hard to copy-paste commands and adds unnecessary confusion (neither `(samcli38)` nor `$` are standard).

I'd much rather get rid of the prompt stuff and just show the commands as-is (with comments to describe behavior or output when needed). Only when _absolutely necessary_ for usability, then showing a single `$` and the output below (as if it was copied from the shell) makes sense.

Reply:
updated